src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinHttpWebSocket.cs

```diff
@@ -688,21 +688,29 @@ private string GetResponseHeaderStringInfo(string headerName)
             }
 
             int charsNeeded = (int)bytesNeeded / sizeof(char);
-            var buffer = new StringBuilder(charsNeeded, charsNeeded);
-
-            results = Interop.WinHttp.WinHttpQueryHeaders(
-                _operation.RequestHandle,
-                Interop.WinHttp.WINHTTP_QUERY_CUSTOM,
-                headerName,
-                buffer,
-                ref bytesNeeded,
-                IntPtr.Zero);
-            if (!results)
+            char[] buffer = new char[charsNeeded];
+            GCHandle pinnedHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
```

Comment:
I switched this to `stackalloc` if `charsNeeded <= 128` stack limit, otherwise allocate a `char[]` on the heap and use `fixed`. The 128 limit is arbitrary, but based on other 256 byte stack limit I saw elsewhere in corefx and the fact that it's very likely the header value will be under 128 chars for the single Sec-WebSocket-Protocol header being queried here (based on the lengths of registered identifiers in the [WebSocket Subprotocol Name Registry](https://www.iana.org/assignments/websocket/websocket.xml); the longest being 55 chars in length).

It might be nice to look into the `stackalloc` with heap-allocated-fallback for `WinHttpHandler`'s  buffer, for cases when the response has a status line + headers length under some stack limit, but I don't know how common that is.